youtube_dl/YoutubeDL.py

```diff
@@ -922,6 +931,8 @@ def iter_playlistitems(format):
 
             x_forwarded_for = ie_result.get('__x_forwarded_for_ip')
 
+            reliably_date_ordered_IEs = ('YoutubeChannelIE, YoutubeUserIE')
```

Comment:
No, no hardcodes here. You are implementing a generic mechanism thus there should be a way to control it on per extractor basis. Or it would be easier to just introduce an option describing playlist order instead of this hack and require user to specify it if he wants such optimization.